applications/GeoMechanicsApplication/custom_utilities/functions_for_parameters.cpp

```diff
@@ -0,0 +1,62 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Anne van de Graaf,
+//                   Marjan Fathian,
+//                   Gennady Markelov
+//
+
+#include "custom_utilities/functions_for_parameters.h" 
+
+namespace Kratos {
+
+    Parameters FunctonsForParameters::ExtractParameters(const Parameters& rSourceParameters,
+                                                        const std::vector<std::string>& rNamesOfParametersToCopy)
+    {
+        auto result = Parameters{};
+        result.CopyValuesFromExistingParameters(rSourceParameters, rNamesOfParametersToCopy);
+        return result;
+    }
+
+    void FunctonsForParameters::AppendParameterNameIfExists(const std::string& rParameterName,
+                                                            const Parameters& rSourceParameters,
+                                                            std::vector<std::string>& rResult)
+    {
+        if (rSourceParameters.Has(rParameterName)) {
+            rResult.emplace_back(rParameterName);
+        }
+    }
+
+    bool FunctonsForParameters::HasTableAttached(const Parameters& rSettings)
+    {
+        if (rSettings["table"].IsNumber()) {
+            return rSettings["table"].GetInt() != 0;
+        }
+
+        KRATOS_ERROR_IF_NOT(rSettings["table"].IsArray()) << "'table' is neither a single number nor an array of numbers";
+
+        const auto& table = rSettings["table"];
+        return std::any_of(table.begin(), table.end(),
+            [](const auto& value) {return value.GetInt() != 0; });
+    }
+
+    bool FunctonsForParameters::HasTableAttached(const Parameters& rSettings, 
+                                                 int component)
+    {
+        if (rSettings["table"][component].IsNumber()) {
+            return rSettings["table"][component].GetInt() != 0;
+        }
+
+        KRATOS_ERROR_IF_NOT(rSettings["table"][component].IsArray()) << "'table' is neither a single number nor an array of numbers";
+
+        const auto& table = rSettings["table"][component];
+        return std::any_of(table.begin(), table.end(),
+            [](const auto& value) {return value.GetInt() != 0; });
+    }
```

Comment:
I doubt whether this is correct. My understanding is that `table` either has a single number or an array of numbers. Now this new overload checks one particular element from the array variant of `table`. By inspecting the original Python code, I don't have the impression that we might have a `table` which is an array, where each element contains an array in turn (which would require two indices to access a number). Therefore, I believe we can simplify this function body to:
```suggestion
        return rSettings["table"][component].GetInt() != 0;
```
Could you please verify whether my understanding is correct? And if yes, please correct the implementation as suggested above. Thanks!

Reply:
Thank you for pointing. I also found only [ "table"].GetInt() and ["table"][0/1/2].GetInt(). The check is removed now.